### PR TITLE
2427 fix spearbit pr 34

### PR DESCRIPTION
**Pull request by rhlsthrm**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/deployments/contracts/contracts/messaging/RootManager.sol [hunk 1]
```diff
@@ -36,6 +36,8 @@ contract RootManager is ProposedOwnable, IRootManager, WatcherClient, DomainInde
 
   event ConnectorRemoved(uint32 domain, address connector, uint32[] domains, address[] connectors, address caller);
 
+  event Log(string message);
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts/messaging/RootManager.sol:39]: Propagate error event maybe?

File: packages/deployments/contracts/contracts_forge/messaging/RootManager.t.sol [hunk 2]
```diff
@@ -253,4 +259,33 @@ contract RootManagerTest is ForgeHelper {
...
+    uint32 domain = uint32(1020);
+    _domains.push(domain);
+    _connectors.push(address(revertConnector));
+
+    vm.prank(owner);
+    _rootManager.addConnector(_domains[20], address(revertConnector));
+
+    bytes32 inboundRoot = keccak256(abi.encode(bytes("test"), 20));
+    vm.prank(address(revertConnector));
+    _rootManager.aggregate(_domains[20], inboundRoot);
```
Comment (LayneHaber) [inline packages/deployments/contracts/contracts_forge/messaging/RootManager.t.sol:279]: Need to check the log event is emitted

File: packages/deployments/contracts/contracts/messaging/RootManager.sol (hunk 1 above)
Comment (jakekidd) [reply packages/deployments/contracts/contracts/messaging/RootManager.sol:39]: `event Event`

Comment (jakekidd) [reply packages/deployments/contracts/contracts/messaging/RootManager.sol:39]: would call it `PropogateFailed` or something? idk

Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/RootManager.sol:39]: something more specific

**Comment by rhlsthrm** [comment]
Updated and fixed.

File: packages/deployments/contracts/contracts_forge/messaging/RootManager.t.sol (hunk 2 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts_forge/messaging/RootManager.t.sol:291]: bump

File: packages/deployments/contracts/contracts/messaging/RootManager.sol (hunk 1 above)
Comment (LayneHaber) [reply packages/deployments/contracts/contracts/messaging/RootManager.sol]: bump

**Review by LayneHaber — CHANGES_REQUESTED** [review]

File: packages/deployments/contracts/contracts/messaging/RootManager.sol (hunk 1 above)
Comment (sanchaymittal) [reply packages/deployments/contracts/contracts/messaging/RootManager.sol]: resolved

**Review by sanchaymittal — APPROVED** [review]

**Review by LayneHaber — APPROVED** [review]

**Comment by LayneHaber** [comment]
Merged into 2152 in #2532

**Pull request merged**